x-pack/plugins/ml/public/embeddables/single_metric_viewer/single_metric_viewer_embeddable_factory.tsx

```diff
@@ -0,0 +1,350 @@
+            throttle((e: { width: number; height: number }) => {
+              // Keep previous container height so it doesn't change the page layout
+              containerHeightRef.current = e.height;
+
+              if (Math.abs(chartWidth - e.width) > 20) {
+                setChartWidth(e.width);
+              }
+            }, RESIZE_THROTTLE_TIME_MS),
+            [chartWidth]
+          );
+
+          const appStateHandler = useCallback(
+            (action: string, payload?: any) => {
+              /**
+               * Empty zoom indicates that chart hasn't been rendered yet,
+               * hence any updates prior that should replace the URL state.
+               */
+
+              switch (action) {
+                case APP_STATE_ACTION.SET_FORECAST_ID:
+                  setSelectedForecastId(payload);
+                  setZoom(undefined);
+                  break;
+
+                case APP_STATE_ACTION.SET_ZOOM:
+                  setZoom(payload);
+                  break;
+
+                case APP_STATE_ACTION.UNSET_ZOOM:
+                  setZoom(undefined);
+                  break;
+              }
+            },
+
+            [setZoom, setSelectedForecastId]
+          );
+
+          if (error) {
+            return (
+              <EuiCallOut
+                title={
+                  <FormattedMessage
+                    id="xpack.ml.singleMetricViewerEmbeddable.errorMessage"
+                    defaultMessage="Unable to load the ML single metric viewer data"
+                  />
+                }
+                color="danger"
+                iconType="warning"
+                css={{ width: '100%' }}
+              >
+                <p>{error}</p>
+              </EuiCallOut>
+            );
+          }
+
+          return (
+            <I18nContext>
+              <KibanaThemeProvider theme={theme}>
+                <KibanaContextProvider
+                  services={{
+                    mlServices: {
+                      ...services[2],
+                    },
+                    ...services[0],
+                    ...services[1],
+                  }}
+                >
+                  <DatePickerContextProvider {...datePickerDeps}>
+                    <Suspense fallback={<EmbeddableLoading />}>
+                      <EuiResizeObserver onResize={resizeHandler}>
+                        {(resizeRef) => (
+                          <div
+                            id={`mlSingleMetricViewerEmbeddableWrapper-${api.uuid}`}
+                            style={{
+                              width: '100%',
+                              overflowY: 'auto',
+                              overflowX: 'hidden',
+                              padding: '8px',
+                            }}
+                            data-test-subj={`mlSingleMetricViewer_${api.uuid}`}
+                            ref={resizeRef}
+                            className="ml-time-series-explorer"
```

Comment:
this class name doesn't exist 
```suggestion
```

Reply:
This class is used in `x-pack/plugins/ml/public/application/timeseriesexplorer/_timeseriesexplorer.scss` - style is shared and used in both `timeseriesexplorer.js` and `timeseriesexplorer_embeddable_chart.js`